doc/man/task-sync.5.in

```diff
@@ -5,21 +5,97 @@ task-sync \- A discussion and tutorial for the various task(1) data
 synchronization capabilities.
 
 .SH INTRODUCTION
-Taskwarrior has several sync options, both external and built in.  If you wish
-to sync your data, choose one method only; mixing methods is going to lead to
-problems.  Each of the methods discussed have their own strengths.
+
+Taskwarrior can "synchronize" your tasks to a server. This has a few benefits:
+.br
+  - Makes your tasks accessible from multiple systems.
+.br
+  - Provides a backup of your tasks.
+.br
+  - Allows internal storage to be reused.
+
+.SH LOCAL SYNCHRONIZATION
+Taskwarrior can sync your tasks locally. While this setup does not make your
+tasks accessible from multiple systems, it does provide a convenient backup
+and allow internal storage to be reused. If you later decide to sync to a
+server, the transition is easy.
+
+Set 'sync.local.server_dir' to the directory where the synchronized data should
+be stored:
+
+    $ task config sync.local.server_dir /path/to/sync
+
+Then run 'task sync' to synchronize to this local "server".
+
+.SH SERVER SYNCHRONIZATION
+Configure each instance of Taskwarrior which you would like to keep in sync
+with the same settings.  You will need the following information from the
+adminisrator of the server:
+.br
+  - The server's URL ("origin", such as "https://tw.example.com")
+.br
+  - A client key ("client_key") identifying your tasks
+.br
+  - An encryption secret ("encryption_secret") used to encrypt and decrypt your
+    tasks.
+
+Configure Taskwarrior with these details:
+
+    $ task config sync.server.origin            <origin>
+    $ task config sync.server.client_key        <client_key>
+    $ task config sync.server.encryption_secret <encryption_secret>
+
+Then run 'task sync' to synchronize the tasks to the server.
+
+To add a new, empty Taskwarrior installation to the set of replicas, such as
+when setting up a new system, run 'task sync' in that installation before
+adding any tasks.
```

Comment:
That's not currently supported. Basically, if you consider the data model to be like Git, then there must be exactly one "root commit". Adding tasks on a new replica would create a new root, and sync'ing that with an existing set of tasks would mean there were then two roots. And, in order to maintain that O(1) storage, once a sync has occurred, that root is no longer stored locally.

We _could_ support this, basically by assuming that all of the local operations don't conflict with operations that have already been sync'd, and just blindly applying them. But I'm not sure "I set up a new laptop, added 2 tasks to it, and then wanted to start syncing with my desktop" is a critical use case. And it would be _much_ harder to support "joining" two sets of tasks that had already been sync'd to different servers. So for the moment, I think it's reasonable to just require that a new replica must be empty.

Reply:
I see. I think this sentence also seems a bit out of place and it's not clear to me what purpose it serves. Didn't we just configure and run `task sync` in the previous sentence? Is this a different scenario?

Also, couldn't "To add a new, empty Taskwarrior installation to the set of replicas" be simplified to "To add a new replica"?